problem/views.py

```diff
@@ -30,24 +41,90 @@ def problem(request):
     return render(request, 'problem/panel.html', {'my_problem':[a,a,a], 'all_problem':[a,a,a,b,b,b]})
 
 def detail(request, problem_id):
-    p = {
-      'pid': problem_id,
-      'title': 'A a+b problem',
-      'description': 'Given a, b, output a+b.',
-      'input': 'a, b <= 100000000',
-      'output': 'a+b',
-      'samp_input': '1 2\n4 5',
-      'samp_output': '3\n9\n',
-      'tag': [''],
-      'testcase': [{'num': 1, 'time': 1, 'memory': 32}, {'num': 2, 'time': 3, 'memory': 100}],
-    }
-    return render(request, 'problem/detail.html', p)
+    problem = Problem.objects.get(pk=problem_id)
```

Comment:
You should check whether the given problem id is valid or not.
If the problem does not exist, write a warning log.